### PR TITLE
Fix for the last closed tab not getting persisted

### PR DESCRIPTION
**Pull request by bomsy**
Associated Issue: #2615 and maybe #2638

### Summary of Changes

* Make sure the `pendingSelectedLocation` is updated

### Test Plan

Tell us a little a bit about how you tested your patch.

Example test plan:

- [x] I had two files opened within the top tab bar.
- [x] I closed both of them (all file tabs closed,
- [x] I refreshed the page
- [x] The last file should be closed.

### Screenshots/Videos 
persist-tabs-fix

### CODE REVIEW

**Comment by jasonLaster** [comment]
@amitzur looks great. Mind encoding those steps in a test :)

* It can be in `integration/tests/tabs` or a new test that is focused on this...
* or if you can put it in a unit test that too :)

File: src/reducers/sources.js [hunk 1]
```diff
@@ -91,10 +91,13 @@ function update(state = State(), action: Action): Record<SourcesState> {
 
     case "CLOSE_TAB":
       availableTabs = removeSourceFromTabList(state.tabs, action.url);
+      location = { sourceId: getNewSelectedSourceId(state, availableTabs) };
```
Comment (jasonLaster) [inline src/reducers/sources.js]: hmm, we might want the line number, no?

Comment (bomsy) [inline src/reducers/sources.js]: Actually we would have a line number because we just closed a tab.

**Comment by bomsy** [comment]
working on tests now!

**Comment by jasonLaster** [comment]
@bomsy I see some unit test errors and flow type errors

**Comment by bomsy** [comment]
yeah...still wip, i just did a quick push so i could work off another laptop

File: src/test/integration/runner.js [hunk 2]
```diff
@@ -107,8 +107,12 @@ describe("Tests", () => {
 
   // expected 2 to equal 1
   xit("source maps bogus", async () => await sourceMapsBogus(ctx));
-
-  it("tabs", async () => await tabs(ctx));
+  xit("tabs - add tabs", async () => await tabs.addTabs(ctx));
+  xit("tabs - reload with tabs", async () => await tabs.reloadWithTabs(ctx));
+  xit(
```
Comment (jasonLaster) [inline src/test/integration/runner.js]: lol

**Pull request merged**